### PR TITLE
chore(components/layout): action button strict mode support

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/551/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/551/storybooks/ag-grid-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/551/storybooks/layout-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/551/storybooks/pages-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/551/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/551/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/551/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/551/playground/)

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 1]
```diff
@@ -44,54 +44,66 @@ export class SkyActionButtonContainerComponent
    */
   @Input()
   public set alignItems(value: SkyActionButtonContainerAlignItemsType) {
-    this._alignItems = value;
+    this.#_alignItems = value;
   }
 
   public get alignItems(): SkyActionButtonContainerAlignItemsType {
-    return this._alignItems || 'center';
+    return this.#_alignItems || 'center';
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: Any getters should be changed to have their logic in the setter instead. We shouldn't be doing logic in getters.

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 2]
```diff
@@ -151,12 +163,14 @@ export class SkyActionButtonContainerComponent
   }
 
   private updateResponsiveClass(): void {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: Private methods should also use the hash privates

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 3]
```diff
@@ -116,8 +128,8 @@ export class SkyActionButtonContainerComponent
   }
 
   public ngOnDestroy(): void {
-    this.ngUnsubscribe.next();
-    this.ngUnsubscribe.complete();
+    this.#ngUnsubscribe.next();
+    this.#ngUnsubscribe.complete();
   }
 
   public onContentChange() {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: I know this wasn't called out before - but care to put return types on functions that are missing one?

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 4]
```diff
@@ -44,54 +44,66 @@ export class SkyActionButtonContainerComponent
...
+    this.#_themeName = value;
     this.updateResponsiveClass();
     this.updateHeight();
   }
 
-  private _alignItems: SkyActionButtonContainerAlignItemsType;
+  #_alignItems: SkyActionButtonContainerAlignItemsType | undefined;
 
-  private _themeName: string;
+  #_themeName: string | undefined;
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts:77]: I wonder if this should default to `default`?

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 5]
```diff
@@ -44,54 +44,66 @@ export class SkyActionButtonContainerComponent
...
   }
 
   public get alignItems(): SkyActionButtonContainerAlignItemsType {
-    return this._alignItems || 'center';
+    return this.#_alignItems || 'center';
   }
 
   @ContentChildren(SkyActionButtonComponent)
-  private actionButtons: QueryList<SkyActionButtonComponent>;
+  private actionButtons: QueryList<SkyActionButtonComponent> | undefined;
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: Content children and view children should be changed to `public`.

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts [hunk 6]
```diff
@@ -44,54 +44,66 @@ export class SkyActionButtonContainerComponent
...
-  private containerRef: ElementRef<any>;
+  private containerRef: ElementRef | undefined;
 
-  private ngUnsubscribe = new Subject<void>();
+  #ngUnsubscribe = new Subject<void>();
 
-  private syncMaxHeightTimeout?: number;
+  #syncMaxHeightTimeout?: number;
 
   private set themeName(value: string) {
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: Setters - if applicable - should also use the hash private. We should have no instances of `private ` when we are done with a component.

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 1 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: fixed

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: done

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: added

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 4 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts:77]: I don't disagree, but I think that change makes more sense if we had a more strongly typed theme name

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 5 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: done

File: libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts (hunk 6 above)
Comment (Blackbaud-CoreyArcher) [reply libs/components/layout/src/lib/modules/action-button/action-button-container.component.ts]: fixed

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request merged**